Rонструкцию try...catch уместно использовать в коде, когда мы работаем с данными из вне. Например, разбираем файл JSON, который получили как ответ на запрос. Или когда мы обрабатываем данные, введенные пользоваетлем... В тех случаях, когда мы не можем быть уверенны в корректности обрабатываемых данных, но нам нужно, чтобы код продалжил выполняться, даже если мы получим ошибку.
